packages/mobile/views/dashboard/drawers/activity/ActivityRouter.svelte

```diff
@@ -1,21 +1,49 @@
 <script lang="typescript">
-    import { activityRoute, ActivityRoute } from '../../../../lib/routers'
-    import { StrongholdUnlock } from '../../../../components'
-    import { claimActivity } from '@core/wallet'
+    import { localize } from '@core/i18n'
+    import { claimActivity, rejectActivity } from '@core/wallet'
+    import { onMount } from 'svelte'
+    import { Confirmation, StrongholdUnlock } from '../../../../components'
+    import { activityRoute, ActivityRoute, activityRouter } from '../../../../lib/routers'
     import { selectedActivity } from '../../../../lib/wallet'
+    import { ActivityDetails } from './views'
 
     export let onClose: () => unknown = () => {}
 
-    function onConfirm(): void {
+    let detailsVisible = false
+
+    function onClaim(): void {
         $selectedActivity && claimActivity($selectedActivity)
-        onClose()
+        handleClose()
+    }
+    function onReject(): void {
+        $selectedActivity && rejectActivity($selectedActivity.id)
+        handleClose()
     }
+    function handleClose(): void {
+        if (detailsVisible) {
```

Comment:
Im not sure I understand why we need this logic here, shouldnt close always be close?

Reply:
The naming is bad 😅 This function checks if you have been on the details view before when you claim or reject. If so, you go back to the details view (For example, if you cancel the rejection). If not (in case of fast claiming/rejecting) the drawer will be closed.

Reply:
I updated/removed this with the new changes